File: 2026/07-July/04-Saturday/solution.go
func minScore(n int, roads [][]int) int {
	graph := make([][][2]int, n+1)

	for _, road := range roads {
		u, v, d := road[0], road[1], road[2]
		graph[u] = append(graph[u], [2]int{v, d})
		graph[v] = append(graph[v], [2]int{u, d})
	}

	visited := make([]bool, n+1)
	ans := math.MaxInt

	var dfs func(int)
	dfs = func(node int) {
		visited[node] = true

		for _, edge := range graph[node] {
			next, dist := edge[0], edge[1]
			if dist < ans {
				ans = dist
			}
			if !visited[next] {
				dfs(next)
			}
		}
	}

	dfs(1)
	return ans
}

File: 2026/07-July/04-Saturday/solution.py
class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:
        graph = defaultdict(list)

        for u, v, d in roads:
            graph[u].append((v, d))
            graph[v].append((u, d))

        visited = set()
        ans = float('inf')

        def dfs(node):
            nonlocal ans
            visited.add(node)

            for nxt, dist in graph[node]:
                ans = min(ans, dist)
                if nxt not in visited:
                    dfs(nxt)

        dfs(1)
        return ans
